common/src/main/java/org/apache/sedona/common/raster/RasterConstructors.java

```diff
@@ -45,6 +55,116 @@ public static GridCoverage2D fromGeoTiff(byte[] bytes) throws IOException {
         return geoTiffReader.read(null);
     }
 
+    /**
+    * Returns a raster that is converted from the geometry provided.
+     * @param geom The geometry to convert
+     * @param raster The reference raster
+     * @param pixelType The data type of pixel/cell of resultant raster
+     * @param value The value of the pixel of the resultant raster
+     * @param noDataValue The noDataValue of the resultant raster
+     *
+     * @return Rasterized Geometry
+     * @throws FactoryException
+    */
+    public static GridCoverage2D asRaster(Geometry geom, GridCoverage2D raster, String pixelType, double value, Double noDataValue) throws FactoryException {
+
+        DefaultFeatureCollection featureCollection = getFeatureCollection(geom, raster.getCoordinateReferenceSystem());
+
+        double[] metadata = RasterAccessors.metadata(raster);
+        // The current implementation doesn't support rasters with properties below
+        // It is not a problem as most rasters don't have these properties
+        // ScaleX < 0
+        if (metadata[4] < 0) {
+            throw new IllegalArgumentException(String.format("ScaleX %f of the raster is negative, it should be positive", metadata[4]));
+        }
+        // ScaleY > 0
+        if (metadata[5] > 0) {
+            throw new IllegalArgumentException(String.format("ScaleY %f of the raster is positive. It should be negative.", metadata[5]));
+        }
+        // SkewX should be zero
+        if (metadata[6] != 0) {
+            throw new IllegalArgumentException(String.format("SkewX %d of the raster is not zero.", metadata[6]));
+        }
+        // SkewY should be zero
+        if (metadata[7] != 0) {
+            throw new IllegalArgumentException(String.format("SkewY %d of the raster is not zero.", metadata[7]));
+        }
+
+        Envelope2D bound = JTS.getEnvelope2D(geom.getEnvelopeInternal(), raster.getCoordinateReferenceSystem2D());
+
+        double scaleX = Math.abs(metadata[4]), scaleY = Math.abs(metadata[5]);
+        if (geom.getGeometryType().equalsIgnoreCase(Geometry.TYPENAME_POINT)) {
+            bound = new Envelope2D(bound.getCoordinateReferenceSystem(), bound.getCenterX() - scaleX * 0.5, bound.getCenterY() - scaleY * 0.5, scaleX, scaleY);
+        }
+
+        int width = (int) bound.getWidth(), height = (int) bound.getHeight();
+
+        if (geom.getGeometryType().equalsIgnoreCase(Geometry.TYPENAME_POINT)) {
+            width = 1;
+            height = 1;
+        } else {
+            // To preserve scale of reference raster
+            width = (int) (width / scaleX);
+            height = (int) (height / scaleY);
+        }
```

Comment:
There are some special treatments for points. I think it is also needed for horizontal and vertical line strings. We also need to add tests for these cases.

Reply:
Actually, any kind of geometry can have a sub-pixel size. I suggest simply checking the width/height of the raster to determine if we need to extend its width or height to at least 1 pixel, so that we don't need special treatments for different types of geometries.